pkg/annotations/annotations.go

```diff
@@ -284,15 +285,27 @@ func (a *Annotations) setPorts(annotations map[string]string) error {
 	return nil
 }
 
-func (a *Annotations) setScheme(annotations map[string]string) error {
+func (a *Annotations) setScheme(annotations map[string]string, ingressNamespace, ingressName string) error {
 	switch {
 	case annotations[schemeKey] == "":
 		return fmt.Errorf(`Necessary annotations missing. Must include %s`, schemeKey)
 	case annotations[schemeKey] != "internal" && annotations[schemeKey] != "internet-facing":
 		return fmt.Errorf("ALB Scheme [%v] must be either `internal` or `internet-facing`", annotations[schemeKey])
 	}
 	a.Scheme = aws.String(annotations[schemeKey])
-	return nil
+	cacheKey := fmt.Sprintf("scheme-%s-%s-%s-%s", config.RestrictScheme, config.RestrictSchemeNamespace, ingressNamespace, ingressName)
+	var err error
+	if item := cacheLookup(cacheKey); item != nil {
+		if item.Value() != nil {
+			err = item.Value().(error)
+		} else {
+			err = nil
+		}
+	} else {
+		err = a.ValidateScheme(ingressNamespace, ingressName)
+		cache.Set(cacheKey, err, time.Minute*60)
```

Comment:
If I'm reading this correctly, we only cache when `ValidateScheme` returns an error (`internal-facing` is rejected)?

What do you think about caching a boolean instead? So that every time the ingress is parsed, and its allowed to be `internet-facing` we don't lookup the configmap again. Perhaps this also warrants bringing the cache down to something like `10` minutes?

Reply:
I'd hope this would also simplify logic, making it more readable.

Thinking it'd become something closer to:

```
if item := cacheLookup(cacheKey); item != nil {
  if item.Value().(bool) {
    return nil  
  }
  return fmt.Errorf("SOME ERROR") 
}

isValid := a.ValidateScheme(ingressNamespace, ingressName)
cache.Set(cacheKey, isValid, time.Minute*10)
if !isValid {
  return fmt.Errorf("SOME ERROR")
}
return nil
``` 

Let me know if that resonates with you @r0fls.

Reply:
updated that just now ^^